provision/acc_provision/apic_provision.py

```diff
@@ -8680,5 +8680,25 @@ def rke_flavor_specific_handling(aci_prefix, data, ports, api_filter_prefix, rke
             data['fvTenant']['children'].append(filt_entry)
 
 
+class ApicKubeConfig_5_2_3_3(ApicKubeConfig):
+
+    ACI_PREFIX = aci_prefix
+
+
+class ApicKubeConfig_5_2_3_4(ApicKubeConfig):
+
+    ACI_PREFIX = aci_prefix
+
+
+class ApicKubeConfig_5_2_3_5(ApicKubeConfig):
+
+    ACI_PREFIX = aci_prefix
+
+
+class ApicKubeConfig_5_2_3_6(ApicKubeConfig):
+
+    ACI_PREFIX = aci_prefix
```

Comment:
Does a new class have to be created, or can we pass the version as an attribute to the class? I would prefer that we dont have to create a new class every time.

Reply:
so this is as per the earlier comment where we inherit the original class and do changes only required for the current version in this class. That was the main idea behind making a new class. So that changes becomes minimum @shastrinator  could you please provide more info as per your comments

Reply:
Yes, if we pass the version as an attribute, then we end up with checks for versions inside apic methods. This way we can override a method in a specific apic version, if there are no changes- then there will be those 2 lines here.

Reply:
We can additionally avoid creating a new apic class for very version by introducing a apic_class_version variable and keeping it the same for all the versions, till a new apic_class_version is needed at which time, then and future versions will use the new class.

Reply:
Since we don't anticipate the APIC class changing actively/at all (ACI won't break backward compatibility). So we can do this on a need basis and have APIC class be a static class regardless of target versions. At the point that it does change, we can do this change.

Reply:
Besides, there is already a get_config method that takes the apic-version as input and does apic-version specific changes.

Reply:
Thanks @shastrinator for providing the inputs. Will be reverting multiple class changes to the original design